src/containerapp/azext_containerapp/_utils.py

```diff
@@ -189,7 +189,10 @@ def await_github_action(cmd, token, repo, branch, name, resource_group_name, tim
     animation.flush()
     animation.tick()
     animation.flush()
-    run = workflow.get_runs()[0]
+    runs = workflow.get_runs()
+    while runs is None or runs.totalCount < 1:
```

Comment:
While this works @StrawnSC to give you some context this issue is that, the call the check list & retrieve the [0] once the GitHub action add is completed. Talking to Sanchit the GitHub actions add Async response returns an operationId & we can poll on the operationId, to validate this is 200 & successful, before we call the list of workflows.